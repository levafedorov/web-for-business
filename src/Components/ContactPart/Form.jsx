import React, {Component} from 'react';
import ErrorInForm from './ErrorInForm';
import PropTypes from "prop-types";




export default class Form extends Component{ 
     

   

   textHandler = (ev) => {
       const target = ev.target;
       this.props.setFormState({[target.name]: target.value}, 'form');
      
   }

    render(){
        const {name, message, phone, city, category, mail, errObj} = this.props;
        
      

return <form action="" method="" className="contact-form">
          <div className="sides">
                <div className="left-side">

                    <div className={`name-container container ${errObj.name ? 'error' : null}`}>
                        {errObj.name ? <ErrorInForm message={errObj.name}/> : null}
                        <input type="text" id="name" name="name" value={name} onChange={this.textHandler} placeholder="Name" className="form-input"/>
                    </div>  

                    <div className={`phone-container container ${errObj.phone ? 'error' : null}`}>
                        {errObj.phone ? <ErrorInForm message={errObj.phone}/> : null}
                        <input type="tel" id="phone-number" name="phone" value={phone} onChange={this.textHandler} className="form-input"/>
                    </div>  

                </div>

                    <div className="right-side">
                        
                            <div className="category-container container">
                                <select name="category" id="category" value={category} onChange={this.textHandler} className="form-input">
                        {this.props.categories.map(item => <option value={item} key={item}>{item}</option>)}
                                </select>
                            </div>    

                            <div className="city-container container">
                                <select name="city" id="city" value={city} onChange={this.textHandler}>
                        {this.props.cities.map(item => <option value={item} key={item}>{item}</option>)}
                                </select>
                            </div> 

                  </div>

          </div>

            <div className={`mail-container container ${errObj.mail? 'error' : null}`}>
            {errObj.mail ? <ErrorInForm message={errObj.mail}/> : null}
            <input type="mail" id="mail" name="mail" placeholder="mail@mail" value={mail} onChange={this.textHandler} className="form-input"/>
            </div>

            <div className="message-container container">
                <textarea name="message" id="message" cols="30" rows="10" placeholder="...message" value={message} onChange={this.textHandler} className="form-input">
                  
               </textarea>
            </div>
    </form>
    }
}

Form.propTypes = {
    form: PropTypes.objectOf(PropTypes.string),
    errObj: PropTypes.objectOf(PropTypes.string)
}

Form.defaultProps = {
    categories: ["First", "Second", "Third"],
    cities:["Praha", "Brno", "Plzen"]
}