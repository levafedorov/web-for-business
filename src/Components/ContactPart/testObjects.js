export const testState = {
    form: {name: '',
        phone: '+420',
        category:'First',
        city:'Praha',
        mail:'',
        message:'',},
    errObj:{
        name: '',
        phone: '',
        category:'',
        city:'',
        mail:'',
        message:''
    }
      }

export  const testFormFilled = {
    form: {name: 'Hello There',
        phone: '+420775645657',
        category:'First',
        city:'Praha',
        mail:'qwery@qwery',
        message:'Message',},
    errObj:{
        name: '',
        phone: '',
        category:'',
        city:'',
        mail:'',
        message:''
    }
        }

export  const testFormErrors = {
    form: {name: 'Hello There',
        phone: '+420775645657',
        category:'First',
        city:'Praha',
        mail:'qwery@qwery',
        message:'Message',},
    errObj:{
        name: 'Error here',
        phone: 'Error here',
        category:'Error here',
        city:'Error here',
        mail:'Error here',
        message:'Error here'
    } }

export const defaultState = {
    form: {name: '',
        phone: '+420',
        category:'First',
        city:'Praha',
        mail:'',
        message:'',},
    errObj:{
        name: '',
        phone: '',
        category:'',
        city:'',
        mail:'',
        message:''
    }
        }
    
export const validObjects = [{
    name: 'Gven Bor≈°',
    phone: '+420445665345',
    category:'First',
    city:'Praha',
    mail:'qwery@gmail.com',
    message:'+71912205321',},
    {name: ' Fedorov  Lev',
    phone: '+783456723322',
    category:'second',
    city:'Brno',
    mail:'fls---@mail.ru',
    message:'Hello there',},
    {name: 'Bilbo Baggins',
    phone: '+3112231234454',
    category:'Third',
    city:'Plzen',
    mail:'asds_asdy@yandex.ru',
    message:'Hello there',
}];

export const invalidObjects = [{
    name: 'Kivi123',
    phone: '+420445665345',
    category:'First',
    city:'Praha',
    mail:'qwery@gmail.com',
    message:'+71912205321',},
    {name: 'Fedorov Lev',
    phone: '+7834534',
    category:'second',
    city:'Brno',
    mail:'mail.ru',
    message:'',},
    {name: 'Nota fa--*',
    phone: '+3112231234454',
    category:'Third',
    city:'Plzen',
    mail:'asdsasdy@yandex.ru',
    message:'', }];
            

export const tryStateObj = { 
    name: 'Name',
    phone: '+420745234',
    category:'Second',
    city:'Brno',
    mail:'qwert@qwery',
    message:'Message here hello'  }