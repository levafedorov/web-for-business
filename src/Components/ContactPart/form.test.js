import React from "react";
import {shallow, mount} from "enzyme";
import Form from "./Form";
import {testState, testFormFilled, testFormErrors} from './testObjects';


const categories = ['Fisrt', 'Second', 'Third'];
const cities = ['Praha', 'Brno', 'Plzen', 'Ostrava', 'Olomouc'];

const props = {...testState, categories, cities};
const propsWithErr = {...testFormErrors};

describe("Form component", () => {
   const setUp = (props) => shallow(<Form {...props}/>); 
   let component;
   let instance;

   test("Should component render", () => {
        component = setUp({...props});
        expect(component).toMatchSnapshot();
   }); 

   test("Should component render with default props", () => {
    component = setUp({...testState});
    expect(component).toMatchSnapshot();
  });

  test("Should component render with filled form", () => {
    component = setUp({...testFormFilled});
    expect(component).toMatchSnapshot();
  });

  test("Should component render with error fields", () => {
    component = setUp({...propsWithErr});
    expect(component).toMatchSnapshot();
  });
  
  describe("Check inputs", () => {
       test("Check all imputs on change", () => {
         const setFormState = jest.fn();
         const component = mount(<Form  setFormState={setFormState} {...props}/>)
         const componentDOM  = component.getDOMNode();
         const allInputElems = [...componentDOM.querySelectorAll('.form-input')];
         const allNames = allInputElems.map(item => item.getAttribute('name'));
         console.log(allNames);

         for(let int = 0; int < allNames.length; int++){
            const input = component.find(`[name='${allNames[int]}']`)
            input.simulate("change", {target: {value: 'check', name: 'check'}});
            expect(setFormState).toHaveBeenCalledWith({check: 'check'}, 'form');
            setFormState.mockRestore();
         }
       });
  });
});