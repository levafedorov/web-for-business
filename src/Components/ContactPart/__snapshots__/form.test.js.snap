// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component Should component render 1`] = `
<form
  action=""
  className="contact-form"
  method=""
>
  <div
    className="sides"
  >
    <div
      className="left-side"
    >
      <div
        className="name-container container null"
      >
        <input
          className="form-input"
          id="name"
          name="name"
          onChange={[Function]}
          placeholder="Name"
          type="text"
        />
      </div>
      <div
        className="phone-container container null"
      >
        <input
          className="form-input"
          id="phone-number"
          name="phone"
          onChange={[Function]}
          type="tel"
        />
      </div>
    </div>
    <div
      className="right-side"
    >
      <div
        className="category-container container"
      >
        <select
          className="form-input"
          id="category"
          name="category"
          onChange={[Function]}
        >
          <option
            key="Fisrt"
            value="Fisrt"
          >
            Fisrt
          </option>
          <option
            key="Second"
            value="Second"
          >
            Second
          </option>
          <option
            key="Third"
            value="Third"
          >
            Third
          </option>
        </select>
      </div>
      <div
        className="city-container container"
      >
        <select
          id="city"
          name="city"
          onChange={[Function]}
        >
          <option
            key="Praha"
            value="Praha"
          >
            Praha
          </option>
          <option
            key="Brno"
            value="Brno"
          >
            Brno
          </option>
          <option
            key="Plzen"
            value="Plzen"
          >
            Plzen
          </option>
          <option
            key="Ostrava"
            value="Ostrava"
          >
            Ostrava
          </option>
          <option
            key="Olomouc"
            value="Olomouc"
          >
            Olomouc
          </option>
        </select>
      </div>
    </div>
  </div>
  <div
    className="mail-container container null"
  >
    <input
      className="form-input"
      id="mail"
      name="mail"
      onChange={[Function]}
      placeholder="mail@mail"
      type="mail"
    />
  </div>
  <div
    className="message-container container"
  >
    <textarea
      className="form-input"
      cols="30"
      id="message"
      name="message"
      onChange={[Function]}
      placeholder="...message"
      rows="10"
    />
  </div>
</form>
`;

exports[`Form component Should component render with default props 1`] = `
<form
  action=""
  className="contact-form"
  method=""
>
  <div
    className="sides"
  >
    <div
      className="left-side"
    >
      <div
        className="name-container container null"
      >
        <input
          className="form-input"
          id="name"
          name="name"
          onChange={[Function]}
          placeholder="Name"
          type="text"
        />
      </div>
      <div
        className="phone-container container null"
      >
        <input
          className="form-input"
          id="phone-number"
          name="phone"
          onChange={[Function]}
          type="tel"
        />
      </div>
    </div>
    <div
      className="right-side"
    >
      <div
        className="category-container container"
      >
        <select
          className="form-input"
          id="category"
          name="category"
          onChange={[Function]}
        >
          <option
            key="First"
            value="First"
          >
            First
          </option>
          <option
            key="Second"
            value="Second"
          >
            Second
          </option>
          <option
            key="Third"
            value="Third"
          >
            Third
          </option>
        </select>
      </div>
      <div
        className="city-container container"
      >
        <select
          id="city"
          name="city"
          onChange={[Function]}
        >
          <option
            key="Praha"
            value="Praha"
          >
            Praha
          </option>
          <option
            key="Brno"
            value="Brno"
          >
            Brno
          </option>
          <option
            key="Plzen"
            value="Plzen"
          >
            Plzen
          </option>
        </select>
      </div>
    </div>
  </div>
  <div
    className="mail-container container null"
  >
    <input
      className="form-input"
      id="mail"
      name="mail"
      onChange={[Function]}
      placeholder="mail@mail"
      type="mail"
    />
  </div>
  <div
    className="message-container container"
  >
    <textarea
      className="form-input"
      cols="30"
      id="message"
      name="message"
      onChange={[Function]}
      placeholder="...message"
      rows="10"
    />
  </div>
</form>
`;

exports[`Form component Should component render with error fields 1`] = `
<form
  action=""
  className="contact-form"
  method=""
>
  <div
    className="sides"
  >
    <div
      className="left-side"
    >
      <div
        className="name-container container error"
      >
        <ErrorInForm
          message="Error here"
        />
        <input
          className="form-input"
          id="name"
          name="name"
          onChange={[Function]}
          placeholder="Name"
          type="text"
        />
      </div>
      <div
        className="phone-container container error"
      >
        <ErrorInForm
          message="Error here"
        />
        <input
          className="form-input"
          id="phone-number"
          name="phone"
          onChange={[Function]}
          type="tel"
        />
      </div>
    </div>
    <div
      className="right-side"
    >
      <div
        className="category-container container"
      >
        <select
          className="form-input"
          id="category"
          name="category"
          onChange={[Function]}
        >
          <option
            key="First"
            value="First"
          >
            First
          </option>
          <option
            key="Second"
            value="Second"
          >
            Second
          </option>
          <option
            key="Third"
            value="Third"
          >
            Third
          </option>
        </select>
      </div>
      <div
        className="city-container container"
      >
        <select
          id="city"
          name="city"
          onChange={[Function]}
        >
          <option
            key="Praha"
            value="Praha"
          >
            Praha
          </option>
          <option
            key="Brno"
            value="Brno"
          >
            Brno
          </option>
          <option
            key="Plzen"
            value="Plzen"
          >
            Plzen
          </option>
        </select>
      </div>
    </div>
  </div>
  <div
    className="mail-container container error"
  >
    <ErrorInForm
      message="Error here"
    />
    <input
      className="form-input"
      id="mail"
      name="mail"
      onChange={[Function]}
      placeholder="mail@mail"
      type="mail"
    />
  </div>
  <div
    className="message-container container"
  >
    <textarea
      className="form-input"
      cols="30"
      id="message"
      name="message"
      onChange={[Function]}
      placeholder="...message"
      rows="10"
    />
  </div>
</form>
`;

exports[`Form component Should component render with filled form 1`] = `
<form
  action=""
  className="contact-form"
  method=""
>
  <div
    className="sides"
  >
    <div
      className="left-side"
    >
      <div
        className="name-container container null"
      >
        <input
          className="form-input"
          id="name"
          name="name"
          onChange={[Function]}
          placeholder="Name"
          type="text"
        />
      </div>
      <div
        className="phone-container container null"
      >
        <input
          className="form-input"
          id="phone-number"
          name="phone"
          onChange={[Function]}
          type="tel"
        />
      </div>
    </div>
    <div
      className="right-side"
    >
      <div
        className="category-container container"
      >
        <select
          className="form-input"
          id="category"
          name="category"
          onChange={[Function]}
        >
          <option
            key="First"
            value="First"
          >
            First
          </option>
          <option
            key="Second"
            value="Second"
          >
            Second
          </option>
          <option
            key="Third"
            value="Third"
          >
            Third
          </option>
        </select>
      </div>
      <div
        className="city-container container"
      >
        <select
          id="city"
          name="city"
          onChange={[Function]}
        >
          <option
            key="Praha"
            value="Praha"
          >
            Praha
          </option>
          <option
            key="Brno"
            value="Brno"
          >
            Brno
          </option>
          <option
            key="Plzen"
            value="Plzen"
          >
            Plzen
          </option>
        </select>
      </div>
    </div>
  </div>
  <div
    className="mail-container container null"
  >
    <input
      className="form-input"
      id="mail"
      name="mail"
      onChange={[Function]}
      placeholder="mail@mail"
      type="mail"
    />
  </div>
  <div
    className="message-container container"
  >
    <textarea
      className="form-input"
      cols="30"
      id="message"
      name="message"
      onChange={[Function]}
      placeholder="...message"
      rows="10"
    />
  </div>
</form>
`;
